### PR TITLE
Exams Yurchenko Illia IM-33

### PR DESCRIPTION
**Pull request by yur-ochka**

### CODE REVIEW

**Comment by DavidkJun** [comment]
I have reviewed all the  commits, and i agree with them.

**Comment by namanica** [comment]
good job has been done!

**Comment by Kiryaoo** [comment]
Agree with all commits

**Comment by alomj** [comment]
you have done a great job!

**Comment by nikama-s** [comment]
I reviewed all commits, good job!

**Comment by nevsetakpohano** [comment]
Agree with all commits, you did a great job!

File: Tasks/filter.js [hunk 1]
```diff
@@ -1,15 +1,13 @@
...
-Filter = (T, t) => {
-  remove = [];
-  for (C of T) {
-    x = T.indexOf(C);
-    if (typeof T[x] !== t) {
-      remove.unshift(x);
-    }
+'use strict';
+const filter = (array, type) => {
+  let remove = [];
```
[the remark this answers is not in the document]
Comment (yur-ochka) [reply Tasks/filter.js]: Done

File: Tasks/intersection.js [hunk 2]
```diff
@@ -1,25 +1,12 @@
...
-      delete object_1[attribute_name];
-      {
-      }
-    }
-    {
-    }
-  }
-  {
+'use strict';
+const intersection = (object1, object2) => {
```
[the remark this answers is not in the document]
Comment (yur-ochka) [reply Tasks/intersection.js:3]: Done

File: Tasks/longest.js [hunk 3]
```diff
@@ -1,28 +1,15 @@
...
-      }
-      {
-        if (i.length >= x) {
-          s = i;
-        } else {
-          i = -1;
+'use strict';
+const longest = (line) => {
+  let longestLength = -1;
+  let longestWord = ['Not found'][0][longestLength++];
```
[the remark this answers is not in the document]
Comment (yur-ochka) [reply Tasks/longest.js]: Done

File: Tasks/reverse.js [hunk 4]
```diff
@@ -1,14 +1,13 @@
...
-    delete DATA[_];
-  }, 1000);
-  return DATA;
+'use strict';
+const reverse = (data) => {
+  let keys = Object.keys(data);
+  for (let key of keys) {
+    const value = data[key];
+    data[value] = key;
+    delete data[key];
```
[the remark this answers is not in the document]
Comment (yur-ochka) [reply Tasks/reverse.js]: Done

File: Tasks/day.js [hunk 5]
```diff
@@ -1,15 +1,12 @@
...
-
-_parse_day_ = (s) => {
-  let i;
-  for (i = 0; i < D.length; i++) {
-    if (s.startsWith(D[i].toLowerCase())) {
-      return i + 1;
-    }
+const parseDay = (nameOfDay) => {
+  for (let i = 0; i < DAYS.length; i++) {
+    if (nameOfDay.startsWith(DAYS[i].toLowerCase())) return i + 1;
```
[the remark this answers is not in the document]
Comment (yur-ochka) [reply Tasks/day.js]: Done

**Pull request closed without merging**